diff --git a/dist/cjs/index.js b/dist/cjs/index.js
index 1bb60cb312c3c3aeef7f0ff135f2bb24d524bc5e..b9850b08835aed4750ff3a3b41a05ef98fa49965 100644
--- a/dist/cjs/index.js
+++ b/dist/cjs/index.js
@@ -10,7 +10,7 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
 };
 Object.defineProperty(exports, "__esModule", { value: true });
 /* eslint-disable @typescript-eslint/no-explicit-any */
-const client_1 = require("@prisma/client");
+const client_1 = require(".prisma/client");
 const driver_js_1 = require("./driver.js");
 /**
  * Define a Prisma extension that adds Kysely query builder methods to the Prisma client
diff --git a/dist/esm/index.js b/dist/esm/index.js
index 84c349067fd0c8dd33b9b66d585f89cb5cdd6896..ada6c7f9be292fc6b1e8dfe5981c4deec2839d5f 100644
--- a/dist/esm/index.js
+++ b/dist/esm/index.js
@@ -8,7 +8,7 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
     });
 };
 /* eslint-disable @typescript-eslint/no-explicit-any */
-import { Prisma } from "@prisma/client";
+import { Prisma } from ".prisma/client";
 import { PrismaDriver } from "./driver.js";
 /**
  * Define a Prisma extension that adds Kysely query builder methods to the Prisma client
